code/modules/mob/living/simple_animal/guardian/types/lightning.dm

```diff
@@ -107,12 +107,10 @@
 						if(ishuman(C))
 							var/mob/living/carbon/human/H = C
 							H.electrocution_animation(20)
-						C.jitteriness += 1000
-						C.do_jitter_animation(jitteriness)
+						C.Jitter(1000)
+						C.do_jitter_animation(1000)
 						C.stuttering += 1
-						spawn(20)
-							if(C)
-								C.jitteriness = max(C.jitteriness - 990, 10)
+						addtimer(CALLBACK(C, /mob.proc/Jitter, 10), 2 SECONDS, TIMER_UNIQUE|TIMER_OVERRIDE)
```

Comment:
This entire section is copypasta from electrocute_act. There should be a specific proc to end this in carbon_defense. This works fine enough as is but this entire section triggers me immensely.

Reply:
is it? i'll maybe see if i can combine them